src/seeds.accounts.cpp

```diff
@@ -542,19 +584,228 @@ void accounts::genesis(name user) // Remove this after Golive
 
 }
 
-void accounts::genesisrep() {
+// void accounts::migraterep(uint64_t account, uint64_t cycle, uint64_t chunksize) {
+//   require_auth(_self);
+//   auto uitr = account == 0 ? users.begin() : users.find(account);
+//   uint64_t count = 0;
+//   while (uitr != users.end() && count < chunksize) {
+//     if (uitr->reputation > 0) {
+//       auto ritr = rep.find(uitr->account.value);
+//       if (ritr != rep.end()) {
+//         rep.modify(ritr, _self, [&](auto& item) {
+//           item.rep = uitr->reputation;
+//         });
+//       } else {
+//         add_rep_item(uitr->account, uitr->reputation);
+//       }
+//     }
+//     uitr++;
+//     count++;
+//   }
+//   if (uitr == users.end()) {
+//     // done
+//     size_set("users.sz"_n, chunksize * cycle + count);
+//   } else {
+//     // recursive call
+//     uint64_t nextaccount = uitr->account.value;
+//     action next_execution(
+//         permission_level{get_self(), "active"_n},
+//         get_self(),
+//         "migraterep"_n,
+//         std::make_tuple(nextaccount, cycle+1, chunksize)
+//     );
+
+//     transaction tx;
+//     tx.actions.emplace_back(next_execution);
+//     tx.delay_sec = 1;
+//     tx.send(nextaccount + 1, _self);
+    
+//   }
+// }
+
+void accounts::rankreps() {
+  rankrep(0, 0, 200);
+}
+
+void accounts::rankrep(uint64_t start_val, uint64_t chunk, uint64_t chunksize) {
   require_auth(_self);
-  auto uitr = users.begin();
 
-  while (uitr != users.end()) {
-    if (uitr -> status == "citizen"_n && uitr -> reputation < 100) {
-      users.modify(uitr, _self, [&](auto& user) {
-        user.reputation = 100;
-      });
+  // DEBUG REMOVE THIS - THIS IS SO THE FUNCTION DOESN"T RUN AMOK
+  //if (chunk > 10) { 
+  //  return;
+  //}
+
+  uint64_t total = sizes.get("rep.sz"_n.value, "user rep size not set").size;
+  uint64_t current = chunk * chunksize;
+  auto rep_by_rep = rep.get_index<"byrep"_n>();
+  auto ritr = start_val == 0 ? rep_by_rep.begin() : rep_by_rep.lower_bound(start_val);
```

Comment:
so we are sorting only this chunk - how do we make sure it's sorted across the chunks?

Reply:
It's sorted by the index - it's always sorted by the system. We only walk over the sorted array and calculate the rank based on the current index vs size of array